Lib/test/test_io/test_general.py

```diff
@@ -921,6 +921,49 @@ def test_RawIOBase_read(self):
         self.assertEqual(rawio.read(2), None)
         self.assertEqual(rawio.read(2), b"")
 
+    def test_exact_RawIOBase(self):
+        rawio = self.MockRawIOWithoutRead((b"ab", b"cd"))
```

Comment:
Maybe I wasn't precise enough. `RawIOBase.read()` is implemented in terms of `readinto`. So, what we want to do is provide a `readinto` method that does something bad as it's given the internal buffer we constructed in C. I don't know *if* it's possible to cause a segfault on main with that approach though.

Stated otherwise, we want some class:

```py
class EvilReadInto(MockRawIOWithoutRead):
    def readinto(self, buf):
        # do something bad with 'buf'
```

And then

```py
r = EvilReadInto(something_here)
r.read()  # on main, this call should crash, but not with this patch
```